modin/pandas/test/test_io.py

```diff
@@ -1281,6 +1281,20 @@ def test_read_parquet(self, make_parquet_file, columns):
             columns=columns,
         )
 
+    @pytest.mark.parametrize("nrows", [40, 80, 160, 320, 640, 1280, 2560, 5120])
+    @pytest.mark.xfail(
+        condition="config.getoption('--simulate-cloud').lower() != 'off'",
+        reason="The reason of tests fail in `cloud` mode is unknown for now - issue #3264",
+    )
+    def test_read_parquet_indexing_by_column(self, make_parquet_file, nrows):
+        # Test indexing into a column of modin with various parquet file row lengths.
+        unique_filename = get_unique_filename(extension="parquet")
+        make_parquet_file(filename=unique_filename, nrows=nrows)
+
+        parquet_df = pd.read_parquet(unique_filename)
+        for col in parquet_df.columns:
+            parquet_df[col]
```

Comment:
I think we should also compare the entire dataframe with `pandas.read_parquet`. If the `df_equals` call reproduces the bug, we can use that instead of indexing on each column.

Reply:
Agree. `DataFrame.__getitem__` may not always trigger data materialization and so potential errors may not be risen. Doing `df_equals` will force modin dataframe to materialize the data by explicitly calling [`._to_pandas()`](https://github.com/modin-project/modin/blob/81cb913b16d4368915f7c3ee2d66d8eba4e750d3/modin/pandas/test/utils. py#L579)